### PR TITLE
Add rb_tainted_str_new2

### PR DESCRIPTION
**Pull request by bjfish**
Used in `sqlite3` and `mysql2`.

### CODE REVIEW

File: src/main/c/cext/ruby.c [hunk 1]
```diff
@@ -965,6 +965,14 @@ VALUE rb_str_new_with_class(VALUE klass, const char *string, long len) {
   return truffle_invoke(truffle_invoke(klass, "class"), "new", rb_str_new(string, len));
 }
 
+VALUE rb_tainted_str_new_cstr(const char *ptr)
+{
```
Comment (chrisseaton) [inline src/main/c/cext/ruby.c]: `{` on the same line.

File: lib/cext/ruby.h [hunk 2]
```diff
@@ -664,10 +664,12 @@ VALUE rb_str_new_cstr(const char *string);
 VALUE rb_str_new_shared(VALUE string);
 VALUE rb_str_new_frozen(VALUE string);
 VALUE rb_str_new_with_class(VALUE klass, const char *string, long len);
+VALUE rb_tainted_str_new_cstr(const char*);
 #define rb_str_new2 rb_str_new_cstr
 #define rb_str_new3 rb_str_new_shared
 #define rb_str_new4 rb_str_new_frozen
 #define rb_str_new5 rb_str_new_with_class
+#define rb_tainted_str_new2 rb_tainted_str_new_cstr
```
Comment (eregon) [inline lib/cext/ruby.h]: Could you add them following each other and after `rb_tainted_str_new`?

File: src/main/c/cext/ruby.c (hunk 1 above)
Comment (bjfish) [inline src/main/c/cext/ruby.c]: Fixed.

File: lib/cext/ruby.h (hunk 2 above)
Comment (bjfish) [inline lib/cext/ruby.h]: Yes, fixed.

**Pull request merged**